awx/playbooks/project_update.yml

```diff
@@ -101,6 +101,50 @@
       tags:
         - update_insights
 
+    - block:
+        - name: Ensure the project archive directory is present
+          file:
+            dest: "{{ project_path|quote }}/.archive"
+            state: directory
+
+        - name: Get archive from url
+          get_url:
+            url: "{{ scm_url|quote }}"
+            dest: "{{ project_path|quote }}/.archive/"
+            url_username: "{{ scm_username|default(omit) }}"
+            url_password: "{{ scm_password|default(omit) }}"
+            force_basic_auth: true
+          register: get_archive
+
+        - name: Unpack archive
```

Comment:
Given the frequency by which updates can run and the impact from unarchiving we could check the checksum at this step... compare it to the current scm version and then NOT do this or run the archiving if they are unchanged.

Reply:
The unarchive step only runs if the get_archive has changed, which should cover this requirement.

Reply:
I think the main issue with the project update frequency, is that `get_url` will "download the file every time and replace the file if the contents change."

So the archive will always be downloaded, but only unpacked if it changes. I expect that downloading is the heavier operation, but there's no avoiding that unless we increase the complexity by trying to handle etag or last-modified headers from the server.

@matburt do you think I should try to implement etag and last-modified support, or shall we just mark this conversation as resolved?